src/nodes/dir.rs

```diff
@@ -107,6 +124,42 @@ impl Dir {
 
         Arc::new(Dir { inode, writable, state: Mutex::from(state) })
     }
+
+    /// Creates a new scaffold directory as a child of the current one.
+    ///
+    /// Errors are all logged, not reported.  The rationale is that a scaffold directory for an
+    /// intermediate path component of a mapping has to always be created, as it takes preference
+    /// over any other on-disk contents.
+    ///
+    /// This is purely a helper function for `map`.  As a result, the caller is responsible for
+    /// inserting the new directory into the children of the current directory.
+    fn new_scaffold_child(&self, underlying_path: Option<&PathBuf>, name: &OsStr, ids: &IdGenerator)
+        -> Arc<Node> {
+        match underlying_path {
+            Some(path) => {
+                let child_path = path.join(name);
+                match fs::symlink_metadata(&child_path) {
+                    Ok(fs_attr) => {
+                        if fs_attr.is_dir() {
+                            Dir::new_mapped(ids.next(), &child_path, &fs_attr, self.writable)
+                        } else {
+                            info!("Mapping clobbers non-directory {} with an immutable directory",
+                                child_path.display());
+                            Dir::new_empty(ids.next(), Some(self))
```

Comment:
Can we factor out this line somehow? I suggest putting `return Dir::new_mapped(...` and then having the remaining cases fall through to the other case.

Reply:
Done. Gets rid of one level of indentation, but not sure this looks much better though.